python/ray/data/_internal/block_batching/iter_batches.py

```diff
@@ -0,0 +1,124 @@
+import collections
+import heapq
+from typing import Iterator, List, Optional, Tuple
+
+from ray.types import ObjectRef
+from ray.data.block import Block, BlockMetadata
+from ray.data._internal.block_batching.interfaces import (
+    Batch,
+    BlockPrefetcher,
+)
+
+
+def bundle_block_refs_to_logical_batches(
+    block_ref_iterator: Iterator[Tuple[ObjectRef[Block], BlockMetadata]],
+    batch_size: Optional[int],
+    drop_last: bool = False,
+) -> Iterator[List[ObjectRef[Block]]]:
+    """Given an iterator of block object references, and their corresponding metadata,
+    bundles the block object references into groups of at least `batch_size`.
+
+
+
+    This function does not do any slicing or creation of actual batch objects.
+    """
+    batch_buffer: List[ObjectRef[Block]] = []
+    buffer_size = 0
+    original_batch_size = batch_size
+
+    if batch_size is None:
+        for block_ref, _ in block_ref_iterator:
+            yield [block_ref]
+    else:
```

Comment:
Can do an early return to avoid needing to indent the long else statement.

Reply:
this function is removed